homework-submission/react-hyf/react-hyf-client/src/components/HomeworkWeek3/HomeworkWeek3.js

```diff
@@ -0,0 +1,50 @@
+import React, { Component } from 'react';
+import Profile from './Profile';
+
+const url = 'https://uinames.com/api/?amount=10';
+
+class HomeworkWeek3 extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { data: [] };
+  }
+
+  componentDidMount() {
+    fetch(url)
+      .then(response => response.json())
+      .then(json => 
+        json.map((elem)=>
```

Comment:
You are setting the state inside the `map` function which will trigger the function 10 times and only set the data in state to 1 item. It's better to use `setState` once, for example; `this.setState({ data: json[0] })`, since the array is already random it does not matter that we take the first element from the array.